pyincore/models/dfr3curve.py

```diff
@@ -88,18 +89,50 @@ def calculate_limit_state_probability(self, hazard_values: dict, curve_parameter
                         conditions_met.append(False)
                         break
                 if all(conditions_met):
-                    probability = evaluateexpression.evaluate(rule["expression"], parameters)
+                    eval_result = evaluateexpression.evaluate(rule["expression"], parameters)
                     break
 
-        if math.isnan(probability):
-            error_msg = "Unable to calculate limit state."
-            if self.rules:
-                error_msg += " Evaluation failed for expression: \n" + json.dumps(self.rules) + "\n"
-                error_msg += "Provided Inputs: \n" + json.dumps(hazard_values) + "\n" + json.dumps(kwargs)
+        if isinstance(eval_result, numpy.ndarray) or isinstance(eval_result, list):  # for repair curves etc.
+            return eval_result
+        else:  # for fragility curves the return is a float
+            if eval_result is None:
+                eval_result = 0.0
+            if math.isnan(eval_result):
+                error_msg = "Unable to calculate limit state."
+                if self.rules:
+                    error_msg += " Evaluation failed for expression: \n" + json.dumps(self.rules) + "\n"
+                    error_msg += "Provided Inputs: \n" + json.dumps(hazard_values) + "\n" + json.dumps(kwargs)
+
+                raise ValueError(error_msg)
+
+            return eval_result
+
+    def solve_curve_for_inverse(self, hazard_values: dict, curve_parameters: dict, **kwargs):
+        """Evaluates expression of the curve by calculating its inverse. Example, ppf for cdf. Only supports cdf() for
+         now. More inverse methods may be added in the future.
+
+        Args:
+            hazard_values (dict): Hazard values. Only applicable to fragilities
+            curve_parameters (dict): Curve parameters.
+            **kwargs: Keyword arguments.
+
+        Returns:
+            any: Result of the evaluated inverse expression. Can be float, numpy.ndarray etc.
 
-            raise ValueError(error_msg)
+        """
+        inverse_rules = []
+        actual_rules = self.rules
+        for rule in self.rules:
+            if ".cdf(" in rule['expression']:
+                new_exp = rule['expression'].replace(".cdf(", ".ppf(")
+                inverse_rules.append({'condition': rule['condition'], 'expression': new_exp})
```

Comment:
Keep repair time as x, and PF as y is very confusing. But i can't think of anything better at the moment. Could we discuss this issue in dev meeting? 
For now let's just keep it as it is.

Reply:
you mean the other way? Keeping PF as x and time as y is confusing? If so, I agree, but don't have a better way to handle it at the moment. That's why I added a comment:
```
# NOTE: Even though the kwarg name is "repair_time", it actually takes  percent of functionality. DFR3
# system currently doesn't have a way to represent the name correctly when calculating the inverse.
```